packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -809,11 +809,24 @@ class _AppBarState extends State<AppBar> {
   }
 
   void _handleScrollNotification(ScrollNotification notification) {
-    if (notification is ScrollUpdateNotification) {
+    if (notification is ScrollUpdateNotification && notification.depth == 0) {
```

Comment:
This is the change from the revert. If there are additional scroll views nested below, they could inaccurately change scrolled under to false, while the outer, main scroll view is actually scrolled under.